src/workerd/api/global-scope.c++

```diff
@@ -307,21 +307,31 @@ void ServiceWorkerGlobalScope::sendTraces(kj::ArrayPtr<kj::Own<Trace>> traces,
     Worker::Lock& lock, kj::Maybe<ExportedHandler&> exportedHandler) {
   auto isolate = lock.getIsolate();
 
-  auto event = jsg::alloc<TraceEvent>(traces);
+  auto tailEvent = jsg::alloc<TailEvent>(traces);
+  auto traceEvent = jsg::alloc<TraceEvent>(traces);
```

Comment:
Do we really need to create two separate events here? The way this is written, it is possible for a worker to expose both a `tail` handler and a `trace` handler, both of which would be called with different event objects that convey the same data. Is that really what we want here?

I'd prefer a compat flag driven approach in which if the compat flag is disabled, we look for and emit the legacy `trace` event using the `TraceEvent` object, and if the flag is enabled, we look for and emit only the new `TailEvent`.  It seems like a much cleaner approach.